File: dashboard/src/app/models/telemetry-state.ts
// Контракт ответа query-service (GET /devices) — должен оставаться совместим с
// ru.telemetry.query.model.TelemetryState на бэкенде.
export interface TelemetryState {
  deviceId: string;
  routeId: number | null;
  lat: number;
  lon: number;
  speedKmh: number | null;
  recordedAt: string;
}

File: query-service/src/main/java/ru/telemetry/query/model/TelemetryState.java
package ru.telemetry.query.model;

import java.time.Instant;

/**
 * Контракт значения в Redis по ключу telemetry:state:{deviceId} — должен оставаться в точности
 * совместим с ru.telemetry.processor.model.TelemetryState на стороне telemetry-processor,
 * который эти значения пишет (общий формат данных, не общий Java-класс).
 */
public record TelemetryState(
        String deviceId,
        Long routeId,
        double lat,
        double lon,
        Double speedKmh,
        Instant recordedAt
) {
}
